Electron/src/RON/components/PingGraph.tsx

```diff
@@ -58,51 +61,37 @@ class PingGraph extends Component<IProps, IState> {
     this.state = {
       ping: [],
       board: "Drive",
-      interval: setInterval(() => this.ICMP("Drive"), 1000),
+      interval: setInterval(() => this.update("Drive"), 1000),
     }
-    this.ICMP = this.ICMP.bind(this)
+    this.update = this.update.bind(this)
     this.boardChange = this.boardChange.bind(this)
   }
 
-  ICMP(device: string): void {
-    // If device is not a network device, it must be a board
-    let deviceInfo = NetworkDevices[device]
-    if (deviceInfo === undefined) {
-      deviceInfo = RovecommManifest[device]
-    }
-    const ip = deviceInfo.Ip
+  componentWillUnmount() {
+    clearInterval(this.state.interval)
+  }
 
-    // Ping command is slightly different for windows/unix
-    let pingCommand = ""
-    if (process.platform === "win32") {
-      pingCommand = `ping -n 1 ${ip}`
+  update(device: string): void {
+    const { ping } = this.state
+    if (ping.length < 10) {
+      if (device in this.props.devices && "ping" in this.props.devices[device] && this.props.devices[device].ping) {
+        ping.push({ x: ping.length + 1, y: this.props.devices[device].ping })
+      } else {
+        ping.push({ x: ping.length + 1, y: -1 })
+      }
     } else {
-      pingCommand = `ping -c 1 ${ip}`
-    }
-
-    exec(pingCommand, (error, stdout, stderr) => {
-      // -1 means not reachable
-      let delay = -1
-      if (!error && !stderr && stdout.indexOf("unreachable") === -1) {
-        // On Windows, failed ping is not necessarily an error
-        // Windows Ex. 'Reply from 8.8.8.8: bytes=32 time=26ms TTL=110'
-        // Unix Ex. '64 bytes from 8.8.8.8: icmp_seq=0 ttl=110 time=387.477 ms'
-        const start = stdout.indexOf("time=") + 5
-        const end = stdout.indexOf("ms")
-        delay = Math.round(parseFloat(stdout.substring(start, end)))
```

Comment:
is there not an error condition anymore at all or am i just missing it?

Reply:
There is "no error condition" because this file no longer does any of its own polling of ping, it gets the ping from pingtool, which does have error handling. This function purely looks at the ping once a second, and updates its local ping list of the last 10 seconds worth of values